lua/lspconfig/jdtls.lua

```diff
@@ -1,33 +1,35 @@
+local os = require 'os'
+
 local configs = require 'lspconfig/configs'
 local util = require 'lspconfig/util'
 local handlers = require 'vim.lsp.handlers'
 
 local server_name = 'jdtls'
 
 local function get_java_executable()
-  local executable = vim.fn.getenv('JAVA_HOME') ~= vim.NIL
-    and vim.fn.expand('$JAVA_HOME/bin/java')
+  local executable = os.getenv('JAVA_HOME') ~= vim.NIL
```

Comment:
I would just allocate a local variable for home here. Also you can test against nil, not vim.nil, and simplify this to if home then...